jax/numpy/lax_numpy.py

```diff
@@ -2345,6 +2345,33 @@ def column_stack(tup):
   return concatenate(arrays, 1)
 
 
+@_wraps(np.choose)
+def choose(a, choices, out=None, mode='raise'):
+  if out is not None:
+    raise NotImplementedError("out argument to jnp.choose()")
+  _check_arraylike('choose', a, *choices)
+  if not issubdtype(_dtype(a), integer):
+    raise ValueError("`a` array must be integer typed")
+  N = len(choices)
+
+  if mode == 'raise':
+    a = core.concrete_or_error(array, a,
+      "The error occurred because jnp.choose was jit-compiled"
+      " with mode='raise'. Use mode='wrap' or mode='clip' instead.")
+    if any((a < 0) | (a >= N)):
+      raise ValueError("invalid entry in choice array")
+  elif mode == 'wrap':
+    a = a % N
+  elif mode == 'clip':
+    a = clip(a, 0, N - 1)
+  else:
+    raise ValueError(f"mode={mode!r} not understood. Must be 'raise', 'wrap', or 'clip'")
+
+  a, *choices = broadcast_arrays(a, *choices)
+  choices = array(choices).reshape(N, -1)
+  return choices[a.ravel(), arange(a.size)].reshape(a.shape)
```

Comment:
Is the `.ravel()` needed here? Wouldn't advanced indexing do what you need?

Reshapes can be expensive on TPU, so it's better to avoid them if you can.

Reply:
Thanks, and Done - it turns out this could be expressed more efficiently using the `indices()` function.